pkg/auth/providers/keycloakoidc/keycloak_provider.go

```diff
@@ -0,0 +1,206 @@
+package keycloakoidc
+
+import (
+	"context"
+	"net/http"
+	"strconv"
+	"strings"
+
+	gooidc "github.com/coreos/go-oidc/v3/oidc"
+	"github.com/pkg/errors"
+	"github.com/rancher/norman/types"
+	v32 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/auth/providers/common"
+	"github.com/rancher/rancher/pkg/auth/providers/oidc"
+	"github.com/rancher/rancher/pkg/auth/tokens"
+	client "github.com/rancher/rancher/pkg/client/generated/management/v3"
+	v3 "github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/types/config"
+	"github.com/rancher/rancher/pkg/user"
+	"github.com/sirupsen/logrus"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	Name      = "keycloakoidc"
+	UserType  = "user"
+	GroupType = "group"
+)
+
+type keycloakoidcProvider struct {
+	keycloakoidcClient *KClient
+	oidcProvider       oidc.Provider
```

Comment:
This could potentially be embedded then the below wrapper functions won't be needed

Reply:
I have now embedded this and do not make the extra functions as I was previously for the wrapper functionality. I am only making the specific functions for the keycloak specific searches